### PR TITLE
[cpe/alps] netcdf-hdf5-parallel

### PR DESCRIPTION
**Pull request by jgphpc**
https://jira.cscs.ch/browse/VCMSA-319 (part2)
Current test codes are serial -> moving to parallel tests, for example:
```
    https://github.com/Unidata/netcdf-c/blob/main/examples/C/simple_xy_wr.c
    https://github.com/Unidata/netcdf-c/blob/main/examples/C/simple_xy_rd.c
    -> https://github.com/Unidata/netcdf-c/blob/main/nc_test4/tst_parallel.c
```

```

[       OK ] ( 1/15) netCDFTest %lang=cpp /2bb65ff7 @hohgant:cpu+PrgEnv-aocc
[       OK ] ( 2/15) netCDFTest %lang=cpp /2bb65ff7 @hohgant:cpu+PrgEnv-cray
[       OK ] ( 3/15) netCDFTest %lang=cpp /2bb65ff7 @hohgant:cpu+PrgEnv-gnu
[       OK ] ( 4/15) netCDFTest %lang=cpp /2bb65ff7 @hohgant:cpu+PrgEnv-nvhpc
[       OK ] ( 5/15) netCDFTest %lang=cpp /2bb65ff7 @hohgant:cpu+PrgEnv-nvidia
[       OK ] ( 6/15) netCDFTest %lang=c /00cc832a @hohgant:cpu+PrgEnv-aocc
[       OK ] ( 7/15) netCDFTest %lang=c /00cc832a @hohgant:cpu+PrgEnv-cray
[       OK ] ( 8/15) netCDFTest %lang=c /00cc832a @hohgant:cpu+PrgEnv-gnu
[       OK ] ( 9/15) netCDFTest %lang=c /00cc832a @hohgant:cpu+PrgEnv-nvhpc
[       OK ] (10/15) netCDFTest %lang=c /00cc832a @hohgant:cpu+PrgEnv-nvidia
[       OK ] (11/15) netCDFTest %lang=f90 /00ff1fae @hohgant:cpu+PrgEnv-aocc
[       OK ] (12/15) netCDFTest %lang=f90 /00ff1fae @hohgant:cpu+PrgEnv-cray
[       OK ] (13/15) netCDFTest %lang=f90 /00ff1fae @hohgant:cpu+PrgEnv-gnu
[       OK ] (14/15) netCDFTest %lang=f90 /00ff1fae @hohgant:cpu+PrgEnv-nvhpc
[       OK ] (15/15) netCDFTest %lang=f90 /00ff1fae @hohgant:cpu+PrgEnv-nvidia
```

### CODE REVIEW

File: checks/libraries/io/netcdf.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    def set_source(self):
+        self.sourcesdir = f'src/netcdf-hdf5parallel/{self.lang}'
+        self.executable = 'wr.exe'
+
+    @run_before('run')
+    def fix_cpe(self):
+        # fix for "GLIBCXX_3.4.29 not found" error:
+        if self.lang == 'cpp' and self.current_environ.name == 'PrgEnv-gnu':
+            self.env_vars = {
+                'LD_PRELOAD': '/opt/cray/pe/gcc/12.2.0/snos/lib64/libstdc++.so'
```
Comment (jgphpc) [inline checks/libraries/io/netcdf.py:37]: 🤮 but I can't find a better way

File: checks/libraries/io/netcdf.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    valid_systems = ['+remote']
+    valid_prog_environs = ['+mpi +netcdf-hdf5parallel']
+    build_system = 'Make'
+    # modules = ['cray-netcdf-hdf5parallel']  # can't use because of VCMSA-344
+    tags = {'production', 'craype', 'health'}
+
+    @run_before('compile')
+    def set_source(self):
+        self.sourcesdir = f'src/netcdf-hdf5parallel/{self.lang}'
+        self.executable = 'wr.exe'
```
Comment (teojgo) [inline checks/libraries/io/netcdf.py:30]: This can go at a class level

File: checks/libraries/io/netcdf.py (hunk 1 above)
Comment (teojgo) [reply checks/libraries/io/netcdf.py:37]: 'LD_PRELOAD': '$GCC_PREFIX/snos/lib64/libstdc++.so'

**Review by teojgo — CHANGES_REQUESTED** [review]

**Review by teojgo — COMMENTED** [review]
Overall it looks good. The only problem that I see is that it's tied to cray specific paths/prg envs so it won't work without a cray environment.

**Comment by jgphpc** [comment]
> Overall it looks good. The only problem that I see is that it's tied to cray specific paths/prg envs so it won't work without a cray environment.

we can fix that when we update the test for uenv

**Review by ekouts — APPROVED** [review]

**Pull request merged**